### PR TITLE
TicketDispenserClient solution changes to support common architecture

### PR DESCRIPTION
**Pull request by ankitit2015**

### CODE REVIEW

File: Parking.Common/ISqlDataAccess.cs [hunk 1]
```diff
@@ -57,6 +57,14 @@ public interface ISqlDataAccess
         /// <summary>
         ///
```
Comment (mohitrawat789) [inline Parking.Common/ISqlDataAccess.cs]: Please remove Empty Comments.

File: Parking.Common/ISqlDataAccess.cs [hunk 2]
```diff
@@ -57,6 +57,14 @@ public interface ISqlDataAccess
         /// <summary>
         /// 
         /// </summary>
-        string ConnectionString { get; set; }
+        /// <param name="command"></param>
+        /// <returns></returns>
+        DataTable ExecuteDataReturningStoredProcedure(SqlCommand command);
+
+        /// <summary>
+        ///
```
Comment (mohitrawat789) [inline Parking.Common/ISqlDataAccess.cs]: Please remove Empty Comments.

File: Parking.Common/Model/TickerDispenserClientSettings.cs [hunk 3]
```diff
@@ -5,13 +5,13 @@
...
-    public class TDClientSetting
+    public class TickerDispenserClientSettings
     {
-        public string TDClientDeviceId { get; set; }        
+        public string DeviceId { get; set; }        
         public string UserId { get; set; }
         public string Password { get; set; }
         public string LongLat { get; set; }
-        public string PLCBoardConnectPort { get; set; }
+        public string PLCBoardPortNumber { get; set; }
```
Comment (mohitrawat789) [inline Parking.Common/Model/TickerDispenserClientSettings.cs]: Has it not been changed to int in last MPS checkin? If not please change it to int

File: Parking.TicketDispenserClient/Forms/VehicleEntry.cs [hunk 4]
```diff
@@ -2,19 +2,18 @@
...
 using System.Threading;
-using Parking.Utilities;
+using System.Threading.Tasks;
 
 namespace Parking.Entry.Forms
 {
     public partial class VehicleEntry : Form
     {
-        private TDClientSetting tdSetting;
+        private TickerDispenserClientSettings settings;
```
Comment (mohitrawat789) [inline Parking.TicketDispenserClient/Forms/VehicleEntry.cs]: please make it readonly .

File: Parking.TicketDispenserClient/Forms/VehicleEntry.cs [hunk 5]
```diff
@@ -2,19 +2,18 @@
...
         private readonly ParkingDatabaseFactory parkingDatabaseFactory;
         private static Ticket ticket;
         private string CompnayName;
```
Comment (mohitrawat789) [inline Parking.TicketDispenserClient/Forms/VehicleEntry.cs]: Company name - spelling incorrect

File: Parking.TicketDispenserClient/Forms/Settings.cs [hunk 6]
```diff
@@ -77,7 +75,7 @@ private void BtnConnectPortClick(object sender, EventArgs e)
         {
             try
             {
-                string portName = tdSetting.PLCBoardConnectPort;
+                string portName = settings.PLCBoardPortNumber;
```
Comment (mohitrawat789) [inline Parking.TicketDispenserClient/Forms/Settings.cs]: PLCBoardPortNumber should be int in TDclientSettings model

File: Parking.Common/Model/TickerDispenserClientSettings.cs (hunk 3 above)
Comment (ankitit2015) [reply Parking.Common/Model/TickerDispenserClientSettings.cs]: PLC Board ports are like 'COM1' etc. It is not string though

File: Parking.TicketDispenserClient/Forms/VehicleEntry.cs (hunk 4 above)
Comment (ankitit2015) [reply Parking.TicketDispenserClient/Forms/VehicleEntry.cs]: Not possible, it is assigned like below:
settings = (TickerDispenserClientSettings)ConfigurationReader.GetConfigurationSettings(Common.Enums.Application.TicketDispenserClient);

File: Parking.TicketDispenserClient/Forms/Settings.cs (hunk 6 above)
Comment (ankitit2015) [reply Parking.TicketDispenserClient/Forms/Settings.cs]: PLC Board ports are like 'COM1' etc. It is not int though

File: Parking.Common/ISqlDataAccess.cs (hunk 2 above)
Comment (ankitit2015) [reply Parking.Common/ISqlDataAccess.cs]: Done

File: Parking.Common/ISqlDataAccess.cs (hunk 1 above)
Comment (ankitit2015) [reply Parking.Common/ISqlDataAccess.cs]: Done

File: Parking.TicketDispenserClient/Forms/VehicleEntry.cs (hunk 5 above)
Comment (ankitit2015) [reply Parking.TicketDispenserClient/Forms/VehicleEntry.cs]: Done

**Comment by ankitit2015** [comment]
Merged

**Pull request merged**